staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -231,11 +232,15 @@ func NewDiffVersion(name string) (*DiffVersion, error) {
 }
 
 func (v *DiffVersion) getObject(obj Object) (runtime.Object, error) {
+	live, merged, err := mask(obj)
```

Comment:
I'm not completely sure yet how to do that, but that's not the way to go. Since this is going to get called twice, you'll end-up running the request twice, which is not ok.